overlord/snapstate/component.go

```diff
@@ -222,17 +391,20 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup ComponentS
 	}
 
 	// security
-	if !compSetup.JointSnapComponentsInstall {
-		setupSecurity := st.NewTask("setup-profiles", fmt.Sprintf(i18n.G("Setup component %q%s security profiles"), compSi.Component, revisionStr))
+	if !compSetup.MultiComponentInstall && setupSecurity == nil {
+		setupSecurity = st.NewTask("setup-profiles", fmt.Sprintf(i18n.G("Setup component %q%s security profiles"), compSi.Component, revisionStr))
 		componentTS.beforeLink = append(componentTS.beforeLink, setupSecurity)
+	}
+
+	if setupSecurity != nil {
 		addTask(setupSecurity)
```

Comment:
addTask is attaching information to the re-used task which many is harmless but is confusing?